legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/auth/GitLabOidcSession.java

```diff
@@ -55,76 +58,66 @@ public boolean equals(Object other)
             return false;
         }
 
-        GitLabOidcSession that = (GitLabOidcSession)other;
+        GitLabOidcSession that = (GitLabOidcSession) other;
         return Objects.equals(this.getUserId(), that.getUserId()) &&
-                Objects.equals(this.getProfile(), that.getProfile()) &&
-                this.getCreationTime().equals(that.getCreationTime()) &&
-                this.tokenManager.equals(that.tokenManager);
+            Objects.equals(this.getProfile(), that.getProfile()) &&
+            this.getCreationTime().equals(that.getCreationTime()) &&
+            this.appInfo.equals(that.appInfo) &&
+            Objects.equals(this.token, that.token);
     }
 
     @Override
     public int hashCode()
     {
-        return getCreationTime().hashCode() ^ getUserId().hashCode() ^ this.tokenManager.hashCode();
-    }
-
-    @Override
-    public Set<GitLabMode> getValidModes()
-    {
-        return this.tokenManager.getValidModes();
-    }
-
-    @Override
-    public boolean isValidMode(GitLabMode mode)
-    {
-        return this.tokenManager.isValidMode(mode);
+        return getCreationTime().hashCode() ^ getUserId().hashCode() ^ this.appInfo.hashCode() ^ Objects.hashCode(this.token);
     }
 
     @Override
-    public boolean gitLabOAuthCallback(GitLabMode mode, String code)
+    public boolean gitLabOAuthCallback(String code)
     {
-        return this.tokenManager.gitLabOAuthCallback(mode, code);
+        GitLabToken token = GitLabToken.newGitLabToken(Constants.TokenType.OAUTH2_ACCESS, GitLabOAuthAuthenticator.newAuthenticator(appInfo).getOAuthTokenFromAuthCode(code));
+        return !token.equals(this.token);
```

Comment:
The token needs to be set on the field.

Reply:
My bad, I messed up my code reading 😢 😢 😢